src/core_ct/core.py

```diff
@@ -94,20 +98,61 @@ def trim(self, axis: int, loc_start: int, loc_end: int | None = None) -> None:
 
         match axis:
             case 0:
-                self.pixel_array = self.pixel_array[
+                new_pixel_array = self.pixel_array[
                     loc_start : len(self.pixel_array) - loc_end
                 ]
             case 1:
-                self.pixel_array = self.pixel_array[
+                new_pixel_array = self.pixel_array[
                     :, loc_start : len(self.pixel_array[0]) - loc_end
                 ]
             case 2:
-                self.pixel_array = self.pixel_array[
+                new_pixel_array = self.pixel_array[
                     :, :, loc_start : len(self.pixel_array[0, 0]) - loc_end
                 ]
             case _:
                 raise ValueError("axis must be a value between 0 and 2 (inclusive)")
 
+        return Core(new_pixel_array, self.pixel_dimensions)
+
+    def trim_by_percent(self, axis: int, percent_start: float,
+                        percent_end: float | None = None) -> Core:
+        """
+        Reduces the dimensions of the core along a specified axis.
+
+        Get a three-dimensional slice of the core scan by trimming off a percent
+        on the requested axis. This function is symmetrical by default.
+
+        Arguments:
+        ---------
+            axis: integer either 0,1,2 specifying which dimension to collapse:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+            percent_on_left: percent to trim off the left side.
+            percent_on_right: percent to trim off the right side.
+
+        Returns:
+        -------
+            A new trimmed core object
+
+        Raises:
+        ------
+            ValueError if axis is a value other than 0, 1, or 2
+        """
+        if percent_end is None:
+            percent_end = percent_start
+
+        if percent_start > 1.0 or percent_end > 1.0:
```

Comment:
This check should be `percent_start + percent_end < 1.0`, error message should be updated accordingly